### PR TITLE
627:  tests on crud kafka connectors

### PR DESCRIPTION
**Pull request by antipova926**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [x] No need to
- [ ] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings(e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    @SneakyThrows
+    public static void afterAll() {
+      Helpers.INSTANCE.apiHelper.deleteTopic(LOCAL, TOPIC_FOR_CONNECTOR);
+    }
+
+    @SneakyThrows
+    @DisplayName("should create a connector")
+    @Test
+    void createConnector() {
+        try {
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java:38]: will that check ever fail, if it all in try catch?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    .isOnPage()
+                    .clickCreateConnectorButton()
+                    .setConnectorConfig(
+                            SINK_CONNECTOR,
+                            FileUtils.getResourceAsString("config_for_create_connector.json")
+                    )
+                    .connectorIsVisible();
+            pages.openConnectorsList(LOCAL).connectorIsVisibleInList(SINK_CONNECTOR, TOPIC_FOR_CONNECTOR);
+        } finally {
+            Helpers.INSTANCE.apiHelper.deleteConnector(LOCAL, FIRST, SINK_CONNECTOR);
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java:49]: it's better to extract in `after`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }
+
+    @SneakyThrows
+    @DisplayName("should update a connector")
+    @Test
+    void updateConnector() {
+        try {
+            Helpers.INSTANCE.apiHelper.createTopic(LOCAL, TOPIC_FOR_UPDATE_CONNECTOR);
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java:58]: maybe extract `Helpers.INSTANCE.apiHelper` to field? Too much wordiness here

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    @SneakyThrows
+    @DisplayName("should delete connector")
+    @Test
+    void deleteConnector() {
+        Helpers.INSTANCE.apiHelper.createTopic(LOCAL, TOPIC_FOR_DELETE_CONNECTOR);
+        Helpers.INSTANCE.apiHelper.sendMessage(LOCAL, TOPIC_FOR_DELETE_CONNECTOR,
+                FileUtils.getResourceAsString("message_content_create_topic.json.json"), " ");
+        Helpers.INSTANCE.apiHelper.createConnector(LOCAL, FIRST,
+                CONNECTOR_FOR_DELETE,
+                FileUtils.getResourceAsString("delete_connector_config.json"));
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java:87]: can we move all prepartion in `beforeAll` ? (in both update/delete connector) so the body of test method would contain  the most relevant details

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        pages.openConnectorsList(LOCAL)
+                .isOnPage()
+                .openConnector(CONNECTOR_FOR_DELETE);
+        pages.openConnectorsView(LOCAL, CONNECTOR_FOR_DELETE)
+                .clickDeleteButton();
+        pages.openConnectorsList(LOCAL).isNotVisible(CONNECTOR_FOR_DELETE);
+        Helpers.INSTANCE.apiHelper.deleteTopic(LOCAL, TOPIC_FOR_DELETE_CONNECTOR);
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java:94]: imho, deletion should be out of test body

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorsList.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Step
+    public ConnectorsList connectorIsVisibleInList(String connectorName, String topicName) {
+        $(By.xpath("//a[text() = '%s']".formatted(connectorName))).shouldBe(Condition.visible);
+        By.xpath("//a[text() = '%s']".formatted(topicName)).refreshUntil(Condition.visible);
+        return this;
+    }
+
+    public ConnectorsList connectorIsUpdatedInList(String connectorName, String topicName) {
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorsList.java:58]: this is the same as `connectorIsVisibleInList`, isn't `updated` part a bit confusing?
may be  just leave  `connectorIsVisibleInList`

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+import static com.codeborne.selenide.Selenide.$;
+import static org.openqa.selenium.Keys.*;
+
+public class ConnectorUpdateView {
+    @Step
+    public ConnectorUpdateView updateConnectorConfig(String configJson) {
+        String os = System.getProperty("os.name");
+        System.out.println(os);
+        if (os.equals("MacOS")){
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java:15]: you're using COMMAND in other lines as well, are you sure, you need `if` only for the first one`sendKeys`?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import org.openqa.selenium.Keys;
+
+import static com.codeborne.selenide.Selenide.$;
+import static org.openqa.selenium.Keys.*;
+
+public class ConnectorUpdateView {
+    @Step
+    public ConnectorUpdateView updateConnectorConfig(String configJson) {
+        String os = System.getProperty("os.name");
+        System.out.println(os);
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java:14]: remove println

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        if (os.equals("MacOS")){
+            $(".ace_text-input").sendKeys(COMMAND, "a");
+        }else{
+            $(".ace_text-input").sendKeys(CONTROL, "a");
+        }
+        $(".ace_text-input").sendKeys(COMMAND, "a");
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java:20]: do you need to click `cmd+a` twice?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        $(".ace_text-input").sendKeys(Keys.BACK_SPACE);
+        $(".ace_text-input").sendKeys(String.valueOf(configJson.toCharArray()));
+        $(".ace_text-input").sendKeys(COMMAND, "a");
+        $(".ace_text-input").sendKeys(SHIFT, TAB);
+        $("div.ace_content").click();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorUpdateView.java:25]: what's that for? First part was for pasting/deleting/replacing, that part is a mystery for me

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorCreateView.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+public class ConnectorCreateView {
+    private static final String path = "ui/clusters/secondLocal/connectors/create_new";
+
+    @Step
+    public ConnectorCreateView setConnectorConfig(String connectName, String configJson) {
+        $(By.xpath("//input[@name='name']")).sendKeys(connectName);
+        $(".ace_text-input").sendKeys(Keys.BACK_SPACE);
+        $(".ace_text-input").sendKeys(Keys.BACK_SPACE);
+        $(".ace_text-input").sendKeys(String.valueOf(configJson.toCharArray()));
+        $(By.xpath("//input[@name='name']")).click();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorCreateView.java:22]: is it just clicking for input?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorCreateView.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+@ExtensionMethod(WaitUtils.class)
+public class ConnectorCreateView {
+    private static final String path = "ui/clusters/secondLocal/connectors/create_new";
+
+    @Step
+    public ConnectorCreateView setConnectorConfig(String connectName, String configJson) {
+        $(By.xpath("//input[@name='name']")).sendKeys(connectName);
+        $(".ace_text-input").sendKeys(Keys.BACK_SPACE);
+        $(".ace_text-input").sendKeys(Keys.BACK_SPACE);
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/ConnectorCreateView.java:20]: why back_space? why twice?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/extensions/FileUtils.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+
+public class FileUtils {
+
+    public static String getResourceAsString(String resourceFileName) throws IOException {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
```
Comment (mikementor) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/extensions/FileUtils.java:12]:
```
File file = new File(classLoader.getResource(“fileTest.txt”).getFile());
    String data = FileUtils.readFileToString(file, “UTF-8");
 ```
 maybe this one?

**Review by mikementor — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java (hunk 1 above)
Comment (antipova926) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/topics/ConnectorsTests.java]: done

**Review by mikementor — APPROVED** [review]

**Pull request merged**